chainconfig/chainconfig_routine.go

```diff
@@ -85,6 +85,25 @@ func (cc *ChainConfigRoutine) run() {
 			} else {
 				// NOTE: errors are logged by the client, no need to log again
 				_ = chainConfigClient.VoteToEnablePendingFeatures(cc.buildNumber)
+				if !isBuildNumberEqual {
+					validatorInfo, err := chainConfigClient.GetValidatorInfo()
+					if err != nil {
+						cc.logger.Error("Failed to retreive build number", "err", err)
+					} else if validatorInfo.Validator == nil {
+						chainConfigClient.SetBuildNumber(0) // Set build number to 0 when validator never vote before
```

Comment:
This isn't going to work right now since `SetValidatorInfo` will error out if you give it a zero build number, but that's not what you want to do here anyway. The way this should work is if `validatorInfo.Validator == nil` then assume that `buildNumber` is zero and therefore call `SetBuildNumber(cc.buildNumber)` just like you would if the build didn't match.